hooks/airtable-conversion-utils.ts

```diff
@@ -18,7 +20,101 @@ function getThumbnailUrl (imageAttachment: any): string|null {
   return largeThumbnail ? largeThumbnail.url : null
 }
 
+class AirtableRecord {
+  protected apiKey: string;
+  private baseId: string;
+  private tableId: string;
+  private view: string;
+
+  constructor (apiKey: string, baseId: string, tableId: string, view: string) {
+    this.apiKey = apiKey
+    this.baseId = baseId
+    this.tableId = tableId
+    this.view = view
+  }
+
+  /**
+   * Get the Airtable field name for a given field ID.
+   *
+   * @param fieldId Field ID
+   * @returns {Promise<string | null>} Field name
+   */
+  public getFieldName (fieldId: string): Promise<string | null> {
+    const base = new Airtable({ apiKey: this.apiKey }).base(this.baseId)
+    let fieldName: string | undefined
+
+    try {
+      return base(this.tableId)
+        .select({
+          fields: [fieldId],
+          view: this.view
+        })
+        .eachPage((records, nextPage) => {
+          for (const record of records) {
+            if (fieldName) {
+              break
+            }
+
+            const recordFields = Object.keys(record.fields)
+
+            if (recordFields.length > 0) {
+              fieldName = recordFields[0]
+            }
+          }
+
+          nextPage()
+        })
+        .then(() => {
+          if (fieldName) {
+            return fieldName
+          } else {
+            return ''
+          }
+        })
+    } catch (error) {
+      console.error(`Error in getFieldName: ${error}`)
+      return Promise.resolve(null)
+    }
+  }
+
+  /**
+   * Get the Airtable field names for all the fields IDs provided.
+   * The object provided must be a map of keys to field IDs.
+   * The returned object will be a map of keys to field names.
+   *
+   * @param fieldIds Field IDs
+   * @returns {Promise<Record<string, string | null>>} Field names mapped to keys
+   */
+  public getAllFieldNames (
+    fieldIds: Record<string, string>
+  ): Promise<Record<string, string | null>> {
+    const fieldNamesPromises = Object.entries(fieldIds).map(
+      ([field, fieldId]) => {
+        return this.getFieldName(fieldId)
+          .then((fieldName) => {
+            if (fieldName) {
+              return { [field]: fieldName }
+            } else {
+              console.warn(`Field name not found for field ID ${fieldId}`)
+            }
+          }).catch((error) => {
+            console.error(`Error in setAllFieldNames: ${error}`)
```

Comment:
@eddybrando quick question: do you run into these errors, locally, when running `npm run generate`

![Screenshot 2023-01-17 at 12 47 40 PM](https://user-images.githubusercontent.com/6276074/213009062-9eea4fb1-c4f8-4aac-89d5-f7f5110ed7e5.png)

Reply:
Thank you for pointing that out! I indeed had an issue with the `view` parameter which I fixed now in https://github.com/Qiskit/qiskit.org/pull/2913/commits/65db9f742392e127708f26da1b151cfec60bbfe8.